qa/tasks/mon_thrash.py

```diff
@@ -237,88 +238,92 @@ def do_thrash(self):
+                time.sleep(self.revive_delay)
+
+                for mon in mons_to_kill:
+                    self.revive_mon(mon)
+                # do more freezes
+                if mons_to_freeze:
+                    for mon in mons_to_freeze:
+                        self.freeze_mon(mon)
+                    self.log('waiting for {delay} secs to unfreeze mons'.format(
+                        delay=self.freeze_mon_duration))
+                    time.sleep(self.freeze_mon_duration)
+                    for mon in mons_to_freeze:
+                        self.unfreeze_mon(mon)
+
+                self.manager.wait_for_mon_quorum_size(len(mons))
+                for m in mons:
                     s = self.manager.get_mon_status(m)
                     assert s['state'] == 'leader' or s['state'] == 'peon'
-                    assert len(s['quorum']) == len(mons)-len(mons_to_kill)
-
-            self.log('waiting for {delay} secs before reviving monitors'.format(
-                delay=self.revive_delay))
-            time.sleep(self.revive_delay)
-
-            for mon in mons_to_kill:
-                self.revive_mon(mon)
-            # do more freezes
-            if mons_to_freeze:
-                for mon in mons_to_freeze:
-                    self.freeze_mon(mon)
-                self.log('waiting for {delay} secs to unfreeze mons'.format(
-                    delay=self.freeze_mon_duration))
-                time.sleep(self.freeze_mon_duration)
-                for mon in mons_to_freeze:
-                    self.unfreeze_mon(mon)
-
-            self.manager.wait_for_mon_quorum_size(len(mons))
-            for m in mons:
-                s = self.manager.get_mon_status(m)
-                assert s['state'] == 'leader' or s['state'] == 'peon'
-                assert len(s['quorum']) == len(mons)
-
-            if self.scrub:
-                self.log('triggering scrub')
-                try:
-                    self.manager.raw_cluster_cmd('scrub')
-                except Exception:
-                    log.exception("Saw exception while triggering scrub")
-
-            if self.thrash_delay > 0.0:
-                self.log('waiting for {delay} secs before continuing thrashing'.format(
-                    delay=self.thrash_delay))
-                time.sleep(self.thrash_delay)
+                    assert len(s['quorum']) == len(mons)
+
+                if self.scrub:
+                    self.log('triggering scrub')
+                    try:
+                        self.manager.raw_cluster_cmd('scrub')
+                    except Exception:
+                        log.exception("Saw exception while triggering scrub")
+
+                if self.thrash_delay > 0.0:
+                    self.log('waiting for {delay} secs before continuing thrashing'.format(
+                        delay=self.thrash_delay))
+                    time.sleep(self.thrash_delay)
+
+        except Exception as e:
```

Comment:
better would be to write a wrapper `do_thrash` method that does the try/except. Rename the existing `do_thrash` to `_do_thrash`.

Reply:
@batrick Fixed. Please check.